cpp/test/Ice/adapterDeactivation/AllTests.cpp

```diff
@@ -66,14 +66,21 @@ allTests(Test::TestHelper* helper)
         cout << "ok" << endl;
     }
 
+    int firstPort = helper->getTestPort(3);
+    int secondPort = helper->getTestPort(4);
     cout << "testing object adapter published endpoints... " << flush;
     {
-        communicator->getProperties()->setProperty("PAdapter.PublishedEndpoints", "tcp -h localhost -p 12345 -t 30000");
+        communicator->getProperties()->setProperty(
+            "PAdapter.PublishedEndpoints",
+            "tcp -h localhost -p " + to_string(firstPort) + " -t 30000");
         Ice::ObjectAdapterPtr adapter = communicator->createObjectAdapter("PAdapter");
         test(adapter->getPublishedEndpoints().size() == 1);
         Ice::EndpointPtr endpt = adapter->getPublishedEndpoints()[0];
-        test(endpt->toString() == "tcp -h localhost -p 12345 -t 30000");
-        Ice::ObjectPrx prx(communicator, "dummy:tcp -h localhost -p 12346 -t 20000:tcp -h localhost -p 12347 -t 10000");
+        test(endpt->toString() == "tcp -h localhost -p " + to_string(firstPort) + " -t 30000");
+        Ice::ObjectPrx prx(
```

Comment:
This is different from the original code. 

Since we're not listening on these ports, keeping 12346 and 12347 should be fine.